pandas/tests/tseries/offsets/test_offsets.py

```diff
@@ -792,6 +784,43 @@ def test_tick_normalize_raises(tick_classes):
         cls(n=3, normalize=True)
 
 
+@pytest.mark.parametrize(
+    "offset_kwargs, ts_arg, expected_arg",
+    [
+        ({"nanoseconds": 1}, 0, "1970-01-01 00:00:00.000000001"),
+        ({"nanoseconds": 5}, 0, "1970-01-01 00:00:00.000000005"),
+        ({"nanoseconds": -1}, 0, "1969-12-31 23:59:59.999999999"),
+        ({"microseconds": 1}, 0, "1970-01-01 00:00:00.000001"),
+        ({"microseconds": -1}, 0, "1969-12-31 23:59:59.999999"),
+        ({"seconds": 1}, 0, "1970-01-01 00:00:01"),
+        ({"seconds": -1}, 0, "1969-12-31 23:59:59"),
+        ({"minutes": 1}, 0, "1970-01-01 00:01:00"),
+        ({"minutes": -1}, 0, "1969-12-31 23:59:00"),
+        ({"hours": 1}, 0, "1970-01-01 01:00:00"),
+        ({"hours": -1}, 0, "1969-12-31 23:00:00"),
+        ({"days": 1}, 0, "1970-01-02 00:00:00"),
+        ({"days": -1}, 0, "1969-12-31 00:00:00"),
+        ({"weeks": 1}, 0, "1970-01-08 00:00:00"),
+        ({"weeks": -1}, 0, "1969-12-25 00:00:00"),
+        ({"months": 1}, 0, "1970-02-01 00:00:00"),
+        ({"months": -1}, 0, "1969-12-01 00:00:00"),
+        ({"years": 1}, 0, "1971-01-01 00:00:00"),
+        ({"years": -1}, 0, "1969-01-01 00:00:00"),
+        ({"minutes": 2, "nanoseconds": 9}, 4, "1970-01-01 00:02:00.000000013"),
```

Comment:
@jbrockmendel Here is the test case you suggest. The test can be passed in my local machine, but the travis is too slow, will look at the result later.

Reply:
nitpick: id make a separate test for this specific case (could have useful name/comment) and avoid the extra clutter in the parametrize here

Reply:
That makes sense, just done.